index.js

```diff
@@ -1,24 +1,29 @@
 'use strict'
+/**
+ * @typedef {import('unified').FrozenProcessor} FrozenProcessor
+ * @typedef {import('unified').RunCallback} RunCallback
+ * @typedef {import('unified').Settings} Settings
```

Comment:
The `Options` type from `hast-util-to-mdast` can be pulled in to give a more precise typing here
```suggestion
 * @typedef {import('hast-util-to-mdast').Options} Settings
```

or 

```suggestion
 * @typedef {import('hast-util-to-mdast').Options} Options
```

Reply:
One issue here is that hast-util-to-mdast seems to need types still, leaving a few `any`s around. Which makes me think I should go over there and offer to help 😁

Reply:
It has types in version 8 https://github.com/syntax-tree/hast-util-to-mdast/releases/tag/8.0.0
That version may not be used here yet?

Reply:
Oh, so it does. And we're on 7 here.

Reply:
Ah, that'd do it.
Either we could leave settings as is.
Or copy https://github.com/syntax-tree/hast-util-to-mdast/blob/42174ccb99308e60c7695170045595d7a151d766/lib/types.js#L26-L32 over, with a note to remove when updating to version 8 (in the next major release)